nixos/modules/services/misc/ersatztv.nix

```diff
@@ -0,0 +1,126 @@
+{
+  config,
+  pkgs,
+  lib,
+  ...
+}:
+
+let
+  inherit (lib)
+    mkIf
+    getExe
+    maintainers
+    mkEnableOption
+    mkOption
+    mkPackageOption
+    ;
+  inherit (lib.types) str path bool;
+  cfg = config.services.ersatztv;
+in
+{
+  options = {
+    services.ersatztv = {
+      enable = mkEnableOption "ErsatzTV";
+
+      package = mkPackageOption pkgs "ersatztv" { };
+
+      user = mkOption {
+        type = str;
+        default = "ersatztv";
+        description = "User account under which ErsatzTV runs.";
+      };
+
+      group = mkOption {
+        type = str;
+        default = "ersatztv";
+        description = "Group under which ErsatzTV runs.";
+      };
+
+      transcodeDir = mkOption {
+        type = path;
+        default = "/var/lib/etv-transcode";
+        description = ''
+          Temporary transcode directory
+        '';
+      };
+
+      configDir = mkOption {
+        type = path;
+        default = "/var/lib/ersatztv";
+        description = ''
+          Directory containing the server configuration files and SQLite Database file
+        '';
+      };
+
+      baseUrl = mkOption {
+        type = str;
+        default = "";
+        description = ''
+          Base URL to support reverse proxies that use paths (e.g. `/ersatztv`)
+        '';
+      };
+
+      openFirewall = mkOption {
+        type = bool;
+        default = false;
+        description = ''
+          Open the default ports in the firewall for the server.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd = {
+      tmpfiles.settings.ersatztvDirs = {
+        "${cfg.transcodeDir}"."d" = {
+          mode = "700";
+          inherit (cfg) user group;
+        };
+        "${cfg.configDir}"."d" = {
+          mode = "700";
+          inherit (cfg) user group;
+        };
+      };
+      services.ersatztv = {
+        description = "ErsatzTV";
+        after = [ "network-online.target" ];
+        wants = [ "network-online.target" ];
+        wantedBy = [ "multi-user.target" ];
+
+        serviceConfig = {
+          Type = "simple";
+          User = cfg.user;
+          Group = cfg.group;
+          UMask = "0077";
+          WorkingDirectory = cfg.configDir;
+          ExecStart = "${getExe cfg.package}";
+          Restart = "on-failure";
+        };
+
+        environment = {
+          ETV_CONFIG_FOLDER = "${cfg.configDir}";
+          ETV_TRANSCODE_FOLDER = "${cfg.transcodeDir}";
+          ETV_BASE_URL = "${cfg.baseUrl}";
+        };
+      };
+    };
+
+    users.users = mkIf (cfg.user == "ersatztv") {
+      ersatztv = {
+        inherit (cfg) group;
+        isSystemUser = true;
+      };
+    };
+
+    users.groups = mkIf (cfg.group == "ersatztv") { ersatztv = { }; };
+
+    networking.firewall = mkIf cfg.openFirewall {
+      # from https://ersatztv.org/docs/user-guide/install#manual-installation-2
+      allowedTCPPorts = [ 8409 ];
```

Comment:
Would you be able to add an option to override the port?

I _think_ that would require editing the `appsettings.json` file so not sure how practical that would be. https://github.com/ErsatzTV/ErsatzTV/blob/c0b5ecd388d60c354e8019e204a0093060edb39c/ErsatzTV/appsettings.json#L31

Reply:
I'll play around with it and see what I can do

Reply:
I played around with setting an environment variable for the port, but it is picked up by the ASP.NET framework too late in the startup so it gets overridden by the default... Looks like I'll need to actually use the config file, how would I go about writing that config file to disk?

Reply:
Oh actually, it might not matter, it looks like it only looks for the config file in the same directory as the executable, I'm not sure that's possible with the way I understand NixOS to work...